core/scripts/cre/environment/environment/billing.go

```
@@ -0,0 +1,241 @@
+			}
+
+			startBillingErr = startBilling(cmd.Context(), timeout, nil)
+			if startBillingErr != nil {
+				waitToCleanUp(timeout)
+				billingRemoveErr := framework.RemoveTestStack(billingplatformservice.DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
+				if billingRemoveErr != nil {
+					fmt.Fprint(os.Stderr, errors.Wrap(billingRemoveErr, manualBillingCleanupMsg).Error())
+				}
+
+				return errors.Wrap(startBillingErr, "failed to start Billing Platform Service")
+			}
+
+			return nil
+		},
+	}
+
+	return cmd
+}
+
+var stopBillingCmd = &cobra.Command{
+	Use:   "stop",
+	Short: "stop the billing platform service",
+	Long:  `stop the billing platform service and clean up resources`,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		return stopBilling()
+	},
+}
+
+func stopBilling() error {
+	setErr := os.Setenv("CTF_CONFIGS", DefaultBillingConfigFile)
+	if setErr != nil {
+		return fmt.Errorf("failed to set CTF_CONFIGS environment variable: %w", setErr)
+	}
+
+	removeCacheErr := removeBillingStateFiles(relativePathToRepoRoot)
+	if removeCacheErr != nil {
+		framework.L.Warn().Msgf("failed to remove cache files: %s\n", removeCacheErr)
+	}
+
+	return framework.RemoveTestStack(billingplatformservice.DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
+}
+
+func removeBillingStateFiles(relativePathToRepoRoot string) error {
+	path := filepath.Join(relativePathToRepoRoot, envconfig.StateDirname, envconfig.BillingStateFilename)
+	absPath, absErr := filepath.Abs(path)
+	if absErr != nil {
+		return errors.Wrap(absErr, "error getting absolute path for billing platform service state file")
+	}
+
+	return os.Remove(absPath)
+}
+
+func startBilling(_ context.Context, cleanupWait time.Duration, setupOutput *environment.SetupOutput) (startupErr error) {
+	// just in case, remove the stack if it exists
+	_ = framework.RemoveTestStack(billingplatformservice.DEFAULT_BILLING_PLATFORM_SERVICE_SERVICE_NAME)
```

why ignore the error?

---

following the same pattern as beholder. cc @Tofel

---

To be honest, this was some copy pasta. I didn't dig into it enough to isolate all the error cases to handle them appropriately such as maybe a connection error to docker where the stack might exist but we can't remove it for some reason. There are multiple other areas where this is done in the local env.